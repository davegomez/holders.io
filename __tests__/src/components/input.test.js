import React from 'react'
import Input from '../../../src/components/input'
import renderer from 'react-test-renderer'

test('Input required component', () => {
  const component = renderer.create(
    <Input
      label='foo'
      required
      isValid={ true }
      type='bar'
      placeholder='baz'
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Input not required component', () => {
  const component = renderer.create(
    <Input
      label='foo'
      required={ false }
      isValid={ false }
      type='bar'
      placeholder='baz'
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

