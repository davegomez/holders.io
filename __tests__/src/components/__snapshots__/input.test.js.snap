// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input not required component 1`] = `
<label
  data-jsx={4212527588}
  htmlFor="foo"
>
  <span
    className={null}
    data-jsx={4212527588}
  >
    foo
  </span>
  <input
    className="invalid"
    data-jsx={4212527588}
    id="foo"
    onBlur={undefined}
    placeholder="baz"
    type="bar"
  />
</label>
`;

exports[`Input required component 1`] = `
<label
  data-jsx={4212527588}
  htmlFor="foo"
>
  <span
    className="required"
    data-jsx={4212527588}
  >
    foo
  </span>
  <input
    className={null}
    data-jsx={4212527588}
    id="foo"
    onBlur={undefined}
    placeholder="baz"
    type="bar"
  />
</label>
`;
